src/axom/core/Array.hpp

```diff
@@ -33,8 +66,8 @@ class Array;
  * \param [in] arr user-supplied Array instance.
  * \return os the updated output stream object.
  */
-template <typename T>
-std::ostream& operator<<(std::ostream& os, const Array<T>& arr);
+template <typename T, IndexType DIM>
```

Comment:
Minor/suggestion: Should the second parameter just be an `int` instead of `axom::IndexType`?
Asking b/c the `IndexType` can sometimes be `int32` and at other times `int64`.

`int` would match what we already have in `primal`, e.g. https://github.com/LLNL/axom/blob/develop/src/axom/primal/geometry/Point.hpp  

See:
https://github.com/LLNL/axom/blob/48ba6490ac471191179145e662b81c091a19a83d/src/axom/core/Types.hpp#L47-L52